lib/file.js

```diff
@@ -235,24 +235,22 @@ class File extends EventEmitter {
         if (chunkSize < maxChunkSize) {
           chunkSize = chunkSize + chunkSizeIncrement
         }
-
-        activeStreams += 1
-        if (activeStreams < maxConnections) {
-          setTimeout(getChunk, 1000)
-        }
       }
 
       if (maxConnections === 1) {
         const r = requestModule(response.g + '/' + apiStart + '-' + (end - 1))
         r.on('error', handleErrors)
+        r.on('response', handleErrors)
         r.pipe(decryptStream)
       } else {
         combined = CombinedStream.create()
 
         // Pass errors from the combined stream to the main stream
         combined.on('error', (err) => stream.emit('error', err))
 
-        getChunk()
+        for (let i = 0; i < maxConnections; i++) {
+            getChunk()
+        }
```

Comment:
When I implemented it I tried to follow the implementation in MEGA's website, where it waited a time before starting new connections. I just tested and now the website open all the connections in the same time, which is what you did so this part LGTM.

Note that now `activeStreams` is a unused variable and should be removed. I'm setting up Travis to check this kind of error.

Reply:
The delay was cause of issue. For fast internet connection when first chunk is downloaded faster than second starts, CombinedStream releases end event, because there are not source streams any more.